analyses/cell-type-consensus/exploratory-notebooks/01-reference-exploration.Rmd

```diff
@@ -0,0 +1,608 @@
+
+One thing I noticed when looking at the labels that have less than the cutoff is that most of them are from scenarios where we have multiple LCAs. 
+Maybe in the case where we have multiple LCAs we are already too broad and we should just eliminate those matches from the beginning. 
+Here I'm looking at the total number of descendants for all terms that show up because a term has multiple LCAs. 
+
+```{r}
+lca_df |> 
+  dplyr::filter(total_lca > 1) |> 
+  dplyr::select(cl_annotation, total_descendants) |> 
+  unique() |>
+  dplyr::arrange(total_descendants)
+```
+
+It looks like most of these terms are pretty broad and are either much higher than the cutoff or right around the cutoff with a few exceptions.
+Things like "bone cell" and "supporting cell" have few descendants, but I would still argue these are very broad terms and not useful. 
+
+I'm going to filter out any matches that show two LCA terms first and then use the cutoff to define labels we would keep. 
+I'll also look to see what cell types we lose when we add this extra filtering step to be sure they are ones that we want to lose. 
+
+```{r}
+# remove any combinations with more than one lca
+filtered_lca_df <- lca_df |> 
+  dplyr::filter(total_lca < 2)
+
+# get a list of cell types to keep based on cutoff 
+updated_celltypes <- filtered_lca_df |> 
+  dplyr::filter(total_descendants <= cutoff) |> 
+  dplyr::pull(cl_annotation) |> 
+  unique()
+
+# which cell types are now missing from the list to keep
+setdiff(celltypes_to_keep, updated_celltypes)
+```
+
+It looks like I am losing a few terms I already said were not specific and then a few other terms, like "hematopoietic precursor cell" and "perivascular cell". I'll look at both of those to confirm we would not want them. 
+
+#### Hematopoietic precursor cell
+
+```{r}
+print_df |> 
+  dplyr::filter(cl_annotation == "hematopoietic precursor cell")
+```
+
+It looks like here we should be keeping these matches because both references have these labels as hematopoietic stem and progenitor cells. 
+I think in the context of pediatric cancer having this label would be helpful, so maybe we shouldn't remove all terms that have 2 LCAs. 
+
+Let's look at what the other LCA is for an example set. 
+
+```{r}
+lca_df |> 
+  dplyr::filter(panglao_ontology == "CL:0000037" & blueprint_ontology == "CL:0000050") |> 
+  dplyr::select(blueprint_annotation_main, blueprint_annotation_fine, panglao_annotation, cl_annotation)
+```
+
+It looks like these terms have both `hematopoietic precursor cell` and `progenitor cell` as LCAs. 
+Personally, I would keep the term for `hematopoietic precursor cell` because I think it's more informative and specific to the type of progenitor cell. 
+
+#### Perivascular cell
+
+```{r}
+print_df |> 
+  dplyr::filter(cl_annotation == "perivascular cell")
+```
+
+I would remove `perivascular cell`, since the cell type labels from PanglaoDB and Blueprint are pretty different from each other. 
+
+## Similarity index 
```

Comment:
I'm not convinced this is as useful as LCA + some simple rules (e.g., total descendants threshold with exceptions).